### PR TITLE
Allow pretty-show 1.11

### LINKED ISSUE #541: nix derivation doesn't build: `Encountered missing or private dependencies: pretty-show >=1.9.5 && <1.10`
I just updated nix-unstable (to commit f601ab37c2f), and currently `hnix` doesn't build:

```
$ nix-build -A haskellPackages.hnix '<nixpkgs>'
these derivations will be built:
  /nix/store/gsvkarm1dkz85lwdp99506vkss1pwqpl-hnix-0.7.1.drv
building '/nix/store/gsvkarm1dkz85lwdp99506vkss1pwqpl-hnix-0.7.1.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/dhnspmn5hc312lm46a98w4n1bz27bp40-ghc-8.8.3.
unpacking sources
unpacking source archive /nix/store/hfsfghz16fanabxzi8hm0vg7bgb6z0c6-hnix-0.7.1.tar.gz
source root is hnix-0.7.1
setting SOURCE_DATE_EPOCH to timestamp 1000000000 of file hnix-0.7.1/tests/TestCommon.hs
patching sources
Run jailbreak-cabal to lift version restrictions on build inputs.
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/setup-package.conf.d -j4 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /build/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/nk5ks453wxlfr9s55wmvh1xa3hs9wwlj-hnix-0.7.1 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --docdir=/nix/store/xa38xhkw1fvwnj1wsfacibfmhhan6ygg-hnix-0.7.1-doc/share/doc/hnix-0.7.1 --with-gcc=gcc --package-db=/build/package.conf.d --ghc-option=-j4 --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --disable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --extra-lib-dirs=/nix/store/49x12wk4l8c0lh97crvwhjbzfxrq1fa2-ncurses-6.2/lib --extra-lib-dirs=/nix/store/hfs34n1yww7ba1m8gbrwrwh6v1y269m6-libffi-3.3/lib --extra-lib-dirs=/nix/store/l9nfr0znv7qibrapym32qj7dxmynfh84-gmp-6.2.0/lib
Using Parsec parser
Configuring hnix-0.7.1...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1022:20 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:475:12 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.0.1.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing or private dependencies:
pretty-show >=1.9.5 && <1.10

builder for '/nix/store/gsvkarm1dkz85lwdp99506vkss1pwqpl-hnix-0.7.1.drv' failed with exit code 1
error: build of '/nix/store/gsvkarm1dkz85lwdp99506vkss1pwqpl-hnix-0.7.1.drv' failed
```

Grepping for `pretty-show` in the derivation ( /nix/store/gsvkarm1dkz85lwdp99506vkss1pwqpl-hnix-0.7.1.drv ) turns up `/nix/store/n1m1dw14zxdqz13c48i1igw8q0adswir-pretty-show-1.10`, so I think it needs to be pinned to an earlier version (or the constraint relaxed).

### PR DESCRIPTION
**Pull request by Anton-Latukha**
Resolves: #541.

<details><summary>Full report</summary>

---
<p>

  1. [The previous release of the package](https://hackage.haskell.org/package/pretty-show-1.9.5) (v 1.9.5,  date `Dec 27 16:54:42 UTC 2018`, [tree](https://github.com/yav/pretty-show/tree/1b484f4b16974a40d80ca43e586337bc90f5d269)).
  2. [Curren release](https://hackage.haskell.org/package/pretty-show-1.10) (v1.10, date `Thu Jan 30 19:08:20 UTC 2020`, [tree](https://github.com/yav/pretty-show/tree/0ff7959238d17547f3d9813be854730e863f01cb)).

  3. Full diff is: https://github.com/yav/pretty-show/compare/1b484f4b16974a40d80ca43e586337bc90f5d269..0ff7959238d17547f3d9813be854730e863f01cb

:heavy_check_mark: Source diff proves [changelog](https://hackage.haskell.org/package/pretty-show-1.10/changelog) statement true:
```text
Changes in 1.10
  * Add support for quasi-quotes, as the preferred way of dealing
    with non-properly showable things.
```

So, an extension in the functionality of the API for printing to humans.

:heavy_check_mark: Tests.

---
</details>

### CODE REVIEW

**Comment by timbertson** [comment]
Any movement on this? It's pretty awkward that `hnix` won't compile in `nix` :/

**Comment by sorki** [comment]
You can use https://github.com/sorki/hnix-overlay meanwhile which has the bounds removed for this and few other offending libs.

**Comment by timbertson** [comment]
Aha, thanks for the tip 👍

**Review by jwiegley — APPROVED** [review]

**Comment by jwiegley** [comment]
@Anton-Latukha @peti Hi! I was listening to you both on Twitch, and your comments that `hnix` could benefit from having more people who can maintain the project and cut releases were spot on. In fact, you've both been invited to become maintainers if you wish, please feel free to unblock things from making progress.

And yes, my interests do take me in more directions than is entirely practical or productive, but sometimes that's how we're built. I'd like to foster an environment where I spawn ideas, and hand off to others who find it interesting and useful, so that things aren't dependent on my terrible task list. I think my wife would have agreed with everything single word that you said. :-)

**Comment by Anton-Latukha** [comment]
@jwiegley @peti 

Thanks.

+====

I have very rustic speech currently, due to a lack of verbal practice. I've understood the level of rustiness, so I started a daily practice of speaking to myself.

I think the continuation would be more hilarious stories 8)

Anyway...

+====

I start to have ideas.

  1. The first idea is to bolt-on: https://packdeps.haskellers.com/
  I currently work on that.
    1.1. I've already subscribed to the RSS feed, so I would track the new versions of dependencies project would need to adjust to, would try to do it myself to not bother others. If I would understand that I am not getting it on time - I would open a report and call others.

  2. Overall I prefer this flow for myself:
    1. Open the report
    2. Discuss
    3. Open a related pull request (maybe close the linked report if the solution is obvious).
    4. Review. Adding ideas to the report. Suggestions for code in PR should go in PR. Ideas on how to solve the topic should go into the report - that is the proper place to gather ideas. If that was the practice of others - that way would prevent people who just have minor arguments on how the issue should be solved, and so - not to block PRs with additional messages of text that in big PRs no-one wants to read through to understand if PR ready. Or if they are really concerned - they would straightly write into PR and try to block it - and that is honest and good.
    5. Merge
    6. Close report that gathered ideas (open other reports from collected ideas).
    7. I am not forcing this flow on people, and I do not expect well-formed specialists to follow it, but I would encourage it on incoming contributors to teach them how to organize the information properly. Anyway, it is useful to teach to open reports before PRs, especially big, because it risky for people to make an epic PR without prior discussion in the report, because all their effort can miss something important that should've been discussed before they share the work. And if it discussed beforehand - epic PRs would be agreed beforehand, so would be merged more easily. That is what I want to show people.

  3. Read API changes, adjust all possible dependencies.
  4. New release to keep everyone happy.
  5. The most important of course is that project must always head toward evaluating and building `nixpkgs` properly - so that should be the main goal all the time. There is no sense in any other work - when that work is not done.

**Comment by jwiegley** [comment]
Sounds like a reasonable workflow to me, @Anton-Latukha, let's do it.

**Comment by Anton-Latukha** [comment]
I would move slow at first, I've built up pretty great theoretical knowledge base, I should become a way faster when I practice it enough to form a right efficient habits and patterns of creativity towards what needs to be done.

John has a pretty good example of an efficient style of writing. I observed of course that you have your style of efficient patterns to be able to write right through the complexities.

**Comment by jwiegley** [comment]
Thankfully we're in no hurry, whatever pace works best for you. :)

**Comment by Anton-Latukha** [comment]
Also currently abandoned `stack2nix` broke my heep "the most modern Haskell tooling pipeline" on my main `NixOS/Haskell/All-HIEs(which internally relies on stack2nix)/GHC&HIE/Emacs/Spacemacs`.

It is (well, currently ... was) pretty great useful tooling. I've written on Reddit/my blog and in my book about it, since before me only couple of people silently achieved to assemble that.

And HIE gets soon deprecated and reimplemented into the full-blown new incarnation of the language engine&IDE.

So for now I would use more classic ways.

**Pull request merged**
